File: Level2.rb
#array1 = [ ‘1’, ‘c’, ‘0’, ‘-1’, ‘5’, ‘b’, ‘100’, ‘37’, ‘a’, ‘20’, ‘18’, ‘12’, ‘0’ ]

array2 = [ '1', 'c', '0', '-1', '5', 'b', '100', '37', 'a', '20', '18', '12', '0' ]
puts array2
puts "If we add 1 to each of the above numbers, and dump the rest, we get......"

array3 = array2.select { |s| s.match(/^\d+$/) }.map(&:next)
array3.insert(2, '0')
puts array3

File: Javascript/Level2.js
input = [ '1', 'c', '0', '-1', '5', 'b', '100', '37', 'a', '20', '18', '12', '0' ] 
numbers = []
letters = []
alert(input);
alert("If we add 1 to each of the above numbers, and dump the rest, we get......");

for ( i=0; i < input.length; i++ ) {
    if (isNaN(input[i]) == true) {
        letters.push(input[i]);
    } else {
        numbers.push(input[i]);
    };
};

for (i = 0; i < numbers.length; i++) {
    (numbers[i]++);
}

alert(numbers);